packages/error/src/index.ts

```diff
@@ -53,3 +53,23 @@ export function newMessage(err: any, newMsg: string): ErrorWithData {
   newErr.message = `${newMsg}: ${newErr.message}`;
   return newErr;
 }
+
+export interface AugmentedErrorOptions {
+  status?: number;
+  info?: string;
+  data?: any;
+  cause?: unknown;
+}
+
+export class AugmentedError extends Error {
```

Comment:
I borrowed the "augment" term from #7968, but I'm open to other ideas.

Reply:
Works for me!